src/react/hooks.js

```diff
@@ -0,0 +1,86 @@
+const { useContext, useEffect, useState, useRef } = require('react')
+const { AtomContext, differ } = require('../react')
+const raf = require('../raf')
+
+const isServer = typeof navigator === 'undefined'
+
+function useAtom (selector, options = {}) {
+  selector = selector || (state => state)
+  const { sync = false, pure = true, observe = !isServer } = options
+  const schedule = sync ? fn => () => fn() : raf
+  const { atom } = useContext(AtomContext)
+  assert(atom, 'No atom found in context, did you forget to wrap your app in <Provider atom={atom} />?')
+  const [mappedProps, setMappedProps] = useState(selector(atom.get()))
+  const ref = useRef({ mappedProps, atom })
+
+  // in case atom is swapped, unobserve
+  if (ref.current.unobserve && atom !== ref.current.atom) {
+    ref.current.unobserve()
+    ref.current.unobserve = null
+  }
```

Comment:
this feels like a reimplementation of what the array we pass into `useEffect` does

Reply:
Hm, I see what you mean, but unfortunately, because I subscribe to atom inline outside useEffect, I also need to make sure I resubscribe correctly. And the `useEffect(.., [atom])` is only used for when component unmounts.

I'll explain why I'm doing this in the comment above..

Reply:
No longer need to do this now that subscribing is in effect